### DISCUSSION TITLE
Cap number of reads fed to hybpiper

### DISCUSSION THREAD

**Discussion by theo-allnutt-bioinformatics** [chrisjackson-pellicle/HybPiper-RBGV#8] [Ideas]
When the number of reads for a sub set of samples may be too large for practical mapping, it would be useful to have a maximum number of reads cap user option, e.g. --max_reads 2e07

Thanks,

Theo

**Comment by chrisjackson-pellicle** [comment]
@theo-allnutt-bioinformatics I like the idea. What do you think the best way to implement it is? I'm wondering about cases where the total number of reads might be large for a given sample, but the percentage of on-target reads might be small (and the on-target percentage might vary between samples, making it difficult to use a standardised cut-off). Capping the raw read number input for mapping has the potential to drop on-target reads beneath critical thresholds and reduce sequence recovery a lot.

**Reply by chrisjackson-pellicle** [reply]
Could always provide it as one of a growing number of 'user beware' options :)

**Reply by theo-allnutt-bioinformatics** [reply]
I thought trimmomatic or fastp (which I normally use for QC) had an option for number of reads to process, but they don't unfortunately. I know bbmap has an option to only map a specified number of reads and that might achieve what I was after - basically to stop HybPiper hanging for a long time on very large read files when processing large batches. On second thoughts, it might be better to do an analysis of read files prior to HybPiper for overrepresentation / entropy using fastp, and then not use Trimmomatic on the HybPiper run. So maybe leave the option out for now. Thanks Chris!
